toolchain/valgrind.py

```diff
@@ -34,6 +35,12 @@ def setUp(self):
         smm = SoftwareManager()
         self.failures = []
 
+        dist = distro.detect()
+        deps = ['gcc', 'make']
+        if dist.name == 'Ubuntu':
+            deps.extend(['g++'])
+        elif dist.name in ['SuSe', 'redhat', 'fedora']:
+            deps.extend(['gcc-c++'])
         for package in ['gcc', 'make']:
```

Comment:
I guess here you need to use "deps" instead of ['gcc', 'make']

for package in deps: